### PR TITLE
memory: connected memory max size to kconfig values

### PR DESCRIPTION
**Pull request by dabekjakub**
Memory sizes passed in kconfig will now have effect on memory map.

Signed-off-by: Jakub Dabek <jakub.dabek@intel.com>

### CODE REVIEW

File: src/platform/intel/cavs/include/cavs/memory.h [hunk 1]
```diff
@@ -49,6 +49,10 @@
 
 #define MAX_MEMORY_SEGMENTS		PLATFORM_HPSRAM_SEGMENTS
 
+#define LP_SRAM_SIZE			(CONFIG_LP_MEMORY_BANKS * 0x10000)
```
Comment (tlauda) [inline src/platform/intel/cavs/include/cavs/memory.h]: `(CONFIG_LP_MEMORY_BANKS * LPSRAM_BANK_SIZE )`

File: src/platform/intel/cavs/include/cavs/memory.h [hunk 2]
```diff
@@ -49,6 +49,10 @@
...
+
+#define HP_SRAM_SIZE			(CONFIG_HP_MEMORY_BANKS * 0x10000)
```
Comment (tlauda) [inline src/platform/intel/cavs/include/cavs/memory.h]: `(CONFIG_HP_MEMORY_BANKS * HPSRAM_BANK_SIZE )`

File: src/arch/xtensa/configs/suecreek_defconfig [hunk 3]
```diff
@@ -3,3 +3,4 @@ CONFIG_CAVS_DMIC=y
 CONFIG_CAVS_SSP=y
 CONFIG_CORE_COUNT=4
 CONFIG_LP_MEMORY_BANKS=1
+CONFIG_HP_MEMORY_BANKS=47
```
Comment (lgirdwood) [inline src/arch/xtensa/configs/suecreek_defconfig]: missing newline at EOF

File: src/platform/cannonlake/include/platform/memory.h [hunk 4]
```diff
@@ -297,10 +296,9 @@
 /* Stack configuration */
 #define SOF_STACK_SIZE		ARCH_STACK_SIZE
 #define SOF_STACK_TOTAL_SIZE	ARCH_STACK_TOTAL_SIZE
-/* SOF_STACK_OFFSET defines how much memory can be power gated */
-#define SOF_STACK_OFFSET	0x150000
+
 /* SOF_STACK_BASE is moved from end of physical memory by offset */
```
Comment (abonislawski) [inline src/platform/cannonlake/include/platform/memory.h]: This comment should be also removed (or changed) if SOF_STACK_OFFSET is not used anymore

**Pull request merged**